kraken: Add new method "has_code_type" in PtRef Grammar (#2728)
hove-io/navitia: documentation/rfc/ptref_grammar.md

| query | signification | note |
|-------|---------------|------|
|`collection.has_code(type, value)`|all the objects of type `collection` that have the code `{type: "type", value: "value"}` in `codes[]`|only the `codes[]` field is used|

> |`collection.has_code_type(type)`|all the objects of type `collection` that have this kind of code `{type: "type"}` in `codes[]`|only the `codes[]` field is used|

Reviewer: ```suggestion
|`collection.has_code_type(type)`|all the objects of type `collection` that has a code with the specific type `{type: "type"}` in `codes[]`|only the `codes[]` field is used|
```
